Remove HMAC with nil secret from documentation (#422)
jwt/ruby-jwt: README.md

* HS512 - HMAC using SHA-512 hash algorithm

```ruby

> # The secret is required. A JWT::DecodeError will be raised if it isn't present.

Reviewer: This is not totally true. You can try to decode with an empty secret. It just needs to be a string. `nil` does not work.

Author: ```suggestion
# The secret must be a string. A JWT::DecodeError will be raised if it isn't provided.
```
